project/Base.scala

```diff
@@ -159,6 +159,23 @@ class Base extends Build {
     def configWithLibs(cfg: Configuration)(dep: ModuleID, deps: ModuleID*): Project =
       withLibs((dep +: deps).map(_ % cfg))
 
+    def withTwitterLib(dep: ModuleID): Project =
+      project.settings(libraryDependencies += {
+        if (doDevelopTwitterDeps.value) {
+          dep.copy(revision = dep.revision+"-SNAPSHOT")
+        } else dep
+      })
+
+    def withTwitterLibs(deps: Seq[ModuleID]): Project =
+      project.settings(libraryDependencies ++= {
+        if (doDevelopTwitterDeps.value) {
+          deps.map { dep => dep.copy(revision = dep.revision+"-SNAPSHOT") }
+        } else deps
+      })
```

Comment:
How about defining `withTwitterLibs` to call out to `withTwitterLib` so that you don't have to do the revision mucking in multiple places?  Something like:

```
def withTwitterLibs(deps: Seq[ModuleID]): Project =
  deps.foldLeft(project) { (_, mod) => withTwitterLib(mod) }
```